plugins/aea-cli-ipfs/aea_cli_ipfs/ipfs_utils.py

```diff
@@ -303,50 +304,53 @@ def remove_unpinned_files(self) -> None:
                 f"Error while performing garbage collection: {str(e)}"
             ) from e
 
-    def download(self, hash_id: str, target_dir: str, fix_path: bool = True) -> str:
+    def download(
+        self, hash_id: str, target_dir: str, fix_path: bool = True, retries: int = 5
+    ) -> str:
         """
         Download dir by it's hash.
 
         :param hash_id: str. hash of file to download
         :param target_dir: str. directory to place downloaded
         :param fix_path: bool. default True. on download don't wrap result in to hash_id directory.
+        :param retries: int. default 5. How often to attempt the download.
         :return: downloaded path
         """
         if not os.path.exists(target_dir):  # pragma: nocover
             os.makedirs(target_dir, exist_ok=True)
 
+        target_path = Path(target_dir) / hash_id
         if os.path.exists(os.path.join(target_dir, hash_id)):  # pragma: nocover
             raise DownloadError(f"{hash_id} was already downloaded to {target_dir}")
 
-        try:
-            self.client.get(hash_id, target_dir)
-        except ipfshttpclient.exceptions.StatusError as e:
-            raise DownloadError(f"error on download {str(e)}") from e
-        downloaded_path = str(Path(target_dir) / hash_id)
-
-        package_path = None
-        if os.path.isdir(downloaded_path):
-            downloaded_files = os.listdir(downloaded_path)
-            if len(downloaded_files) > 0:
-                package_name, *_ = os.listdir(downloaded_path)
-                package_path = str(Path(target_dir) / package_name)
-
-        if package_path is None:
+        while not target_path.exists() and retries:
+            retries -= 1
+            try:
+                self.client.get(hash_id, target_dir)
+            except ipfshttpclient.exceptions.StatusError as e:
+                logging.error(f"error on download of {hash_id}: {e}")
+        if not target_path.exists():
+            raise DownloadError(f"Failed to download: {hash_id}")
```

Comment:
retry, raise only if file still does not exist after attempting download #retries amount of times